singer_sdk/exceptions.py

```diff
@@ -98,6 +98,20 @@ def __init__(self, message: str, response: requests.Response | None = None) -> N
         self.response = response
 
 
+class IgnorableAPIResponseCodeException(Exception):
+    """Raised when an API response code is ignorable."""
+
+    def __init__(self, status_code: int | None = None) -> None:
+        """Extends the default with the ignored response's status code as an attribute.
+
+        Args:
+            status_code (int): Ignored status code.
+        """
+        message = f"Ignorable API response code: {status_code}"
```

Comment:
Hi @rawwar, thanks for implementing this. Looks like I tried over a year ago and never finished it for some mysterious reason :sweat_smile: I came back here looking for the same functionality again.

For `tap-github`, we are facing a bunch of situations where this would be very useful. But github works very hard to make their API a pain to use, so just mentioning the HTTP status code in the exception message isn't enough, it would be helpful to be able to include some extra text. To give a concrete example, trying to fetch the contributors list for the linux kernel, github returns:
`HTTP status 403 - Forbidden`
```json
{
  "message": "The history or contributor list is too large to list contributors for this repository via the API.",
  "documentation_url": "https://docs.github.com/rest/reference/repos#list-repository-contributors"
}
```
But in some other cases they also use that same error code for other problems. So I'd like to be able to write something like:
```python
raise IgnorableException(status_code=403, error_message="contributor list is too large")
```
That second argument could be optional, so it defaults to what you're suggesting above.

Reply:
Thank you. I'll do what you mentioned. Will be away for 4 days. But, once I am back.. will update the code

Reply:
updated